web/pdf_thumbnail_view.js

```diff
@@ -191,7 +190,16 @@ var PDFThumbnailView = (function PDFThumbnailViewClosure() {
       this.div.setAttribute('data-loaded', true);
       this.ring.appendChild(canvas);
 
-      return canvas.getContext('2d');
+      var ctx = canvas.getContext('2d');
+      var outputScale = getOutputScale(ctx);
+      canvas.width = this.canvasWidth * outputScale.sx;
+      canvas.height = this.canvasHeight * outputScale.sy;
+      canvas.style.width = this.canvasWidth + 'px';
+      canvas.style.height = this.canvasHeight + 'px';
+      if (!unscaled) {
```

Comment:
Please change this line to `if (!unscaled && outputScale.scaled) {`, in order to avoid calling `ctx.scale` when it's not necessary (see https://github.com/mozilla/pdf.js/blob/master/web/ui_utils.js#L88-L90).